sim/dex-abilities.ts

```diff
@@ -50,3 +49,56 @@ export class Ability extends BasicEffect implements Readonly<BasicEffect> {
 		}
 	}
 }
+
+export class DexAbilities {
+	readonly dex: ModdedDex;
+	readonly abilityCache = new Map<ID, Ability>();
+
+	constructor(dex: ModdedDex) {
+		this.dex = dex;
+	}
+
+	get(name: string | Ability = ''): Ability {
+		if (name && typeof name !== 'string') return name;
+
+		const id = toID(name);
+		return this.getByID(id);
+	}
+
+	getByID(id: ID): Ability {
+		let ability = this.abilityCache.get(id);
+		if (ability) return ability;
+		if (this.dex.data.Aliases.hasOwnProperty(id)) {
+			ability = this.get(this.dex.data.Aliases[id]);
+			if (ability.exists) {
+				this.abilityCache.set(id, ability);
+			}
+			return ability;
+		}
+		if (id && this.dex.data.Abilities.hasOwnProperty(id)) {
+			const abilityData = this.dex.data.Abilities[id] as any;
+			const abilityTextData = this.dex.getDescs('Abilities', id, abilityData);
+			ability = new Ability({
+				name: id,
+				...abilityData,
+				...abilityTextData,
+			});
+			if (ability.gen > this.dex.gen) {
+				(ability as any).isNonstandard = 'Future';
+			}
+			if (this.dex.currentMod === 'letsgo' && ability.id !== 'noability') {
+				(ability as any).isNonstandard = 'Past';
+			}
+			if ((this.dex.currentMod === 'letsgo' || this.dex.gen <= 2) && ability.id === 'noability') {
+				(ability as any).isNonstandard = null;
+			}
+		} else {
+			ability = new Ability({
+				id, name: id, exists: false,
+			});
+		}
+
+		if (ability.exists) this.abilityCache.set(id, ability);
+		return ability;
```

Comment:
```suggestion
		if (this.dex.data.Aliases.hasOwnProperty(id)) {
			ability = this.get(this.dex.data.Aliases[id]);
		} else if (id && this.dex.data.Abilities.hasOwnProperty(id)) {
			const abilityData = this.dex.data.Abilities[id] as any;
			const abilityTextData = this.dex.getDescs('Abilities', id, abilityData);
			ability = new Ability({
				name: id,
				...abilityData,
				...abilityTextData,
			});
			if (ability.gen > this.dex.gen) {
				(ability as any).isNonstandard = 'Future';
			}
			if (this.dex.currentMod === 'letsgo' && ability.id !== 'noability') {
				(ability as any).isNonstandard = 'Past';
			}
			if ((this.dex.currentMod === 'letsgo' || this.dex.gen <= 2) && ability.id === 'noability') {
				(ability as any).isNonstandard = null;
			}
		} else {
			ability = new Ability({
				id, name: id, exists: false,
			});
		}

		if (ability.exists) this.abilityCache.set(id, ability);
		return ability;
```
Why mix else-if and returning from an `if` statement? This seems cleaner.

Reply:
Actually are the two alias checks intended to do different things?

Reply:
I only see the one alias check? There's the alias branch, the extant-ability branch, and the nonexistent-ability branch.

Reply:
Oh yeah I see that now.